File: greedy/1439-뒤집기_실5.kt
fun main(args: Array<String>){
    val sc = Scanner(System.'in')
    eng = sc.next()
    print(result(eng))

}

fun result(s : String) : Int{
    var one = 0
    var zero = 0
    stack = mutableListOf(s[0])
    if s[0] == "0" one++ else zero ++

    if s.length == s.count {c -> c == '1' or c == '0'} return 0

    for(i in 1..s.length){
        if s[i] == "1" and stack[0] == "0"{
            stack.removeAt(0)
            one++
            stack.add("1")
        }
            
        
        if s[i] == "0" and stack[0] == "1"{
            stack.removeAt(0)
            zero++
            stack.add("0")
        }

    }

    return min(one,zero)
}

File: greedy/1439-뒤집기_실5.py
import sys
read = sys.stdin.readline
eng = read()
stack = [eng[0]]
one, zero = 0, 0
if eng[0] == "0":
    zero += 1
else:
    one += 1

if len(eng)-1 == eng.count("1") or len(eng)-1 == eng.count("0"):
    print(0)
else:
    for i in range(1,len(eng)):
        if eng[i] == "1":
            if stack[0] == "0":
                stack.pop()
                one += 1
                stack.append("1")
        if eng[i] == "0":
            if stack[0] == "1":
                stack.pop()
                zero += 1
                stack.append("0")
    print(min(one,zero))

    
